OnlineShop/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using DataAccess.EF;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using DataAccess.EF.Models;
+
+namespace OnlineShop.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CategoriesController : ControllerBase
+    {
+        public CategoriesController(IUnitOfWork uow)
+        {
+            _uow = uow;
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<Category>> GetCategories()
+        {
+            return _uow.CategoryRepository.Get().ToList();
+        }
+
+        [HttpGet("{id:int}")]
+        public ActionResult<Category> GetCategory(int id)
+        {
+            var category = _uow.CategoryRepository.Find(id);
+
+            if (category == null)
+            {
+                return NotFound();
+            }
+
+            return category;
+        }
+
+        [HttpPut("{id:int}")]
+        public IActionResult PutCategory(int id, Category? category)
+        {
+            if (category == null || id != category.Id) return BadRequest();
```

Comment:
again. Why do you need this id? if you are extending 'categories' resource, not one category